test/gtest/uct/ib/test_ib_xfer.cc

```diff
@@ -129,13 +133,86 @@ UCS_TEST_P(uct_p2p_mix_test_alloc_methods, mix1000)
     run(1000);
 }
 
-UCS_TEST_P(uct_p2p_mix_test_alloc_methods, mix1000_multithreaded,
-           "REG_MT_THRESH=1", "REG_MT_CHUNK=1K", "REG_MT_BIND=y")
+UCT_INSTANTIATE_IB_TEST_CASE(uct_p2p_mix_test_alloc_methods)
+
+
+class uct_p2p_mix_test_mt : public uct_p2p_mix_test {
+protected:
+    bool is_page_size_aligned(const mapped_buffer &buffer)
+    {
+        return ucs_padding((size_t)buffer.ptr(), ucs_get_page_size()) == 0;
+    }
+
+    virtual void check_buffers(const mapped_buffer &sendbuf,
+                               const mapped_buffer &recvbuf) override
+    {
+        if (!is_page_size_aligned(sendbuf) || !is_page_size_aligned(recvbuf)) {
+            UCS_TEST_SKIP_R(
+                    "MT registration for unaligned buffers is unsupported");
+        }
+
+        uint32_t mt_flag = UCT_IB_MEM_MULTITHREADED;
+        ASSERT_TRUE(((uct_ib_mem_t*)sendbuf.memh())->flags & mt_flag);
+        ASSERT_TRUE(((uct_ib_mem_t*)recvbuf.memh())->flags & mt_flag);
+    }
+
+    bool check_md_flags()
+    {
+#if HAVE_DEVX
+        auto *ib_mlx5_md = ucs_derived_of(sender().md(), uct_ib_mlx5_md_t);
```

Comment:
1. do this after checking md is mlx5
2. check by md ops function pointer (need mlx5 devx md) rather than by MLX5 PRM flag

Reply:
I changed it to the following check:
```c
        auto *ib_md = ucs_derived_of(sender().md(), uct_ib_md_t);
        if (strcmp(ib_md->name, UCT_IB_MD_NAME(mlx5))) {
            return false;
        }
```

I think it should be enough, isn't it?